File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
      vector<int> ans(nums.size());
      ans[0]=1;
      int prefixp=1;
      int suffixp=1;
      for(int i=1;i<nums.size();i++)
      {
       prefixp *= nums[i-1];
        ans[i]= prefixp;
      }
      for(int i=nums.size()-2;i>-1;i--)
      {
       suffixp *= nums[i+1];
        ans[i]= ans[i]*suffixp;
      }
      return ans;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans = new int[nums.length];
        int prefix =1;
        int suffix =1;
        for(int i=0;i<nums.length;i++){
            ans[i]=prefix;
            prefix*=nums[i];
        }
        for(int i=nums.length-1;i>-1;i--){
            ans[i]=suffix*ans[i];
            suffix*=nums[i];
        }
        return ans;
    }
}